packages/TorneloScoresheet/src/pages/TablePairingSelection.tsx

```diff
@@ -1,106 +1,174 @@
-  
-  if (appModeState.mode !== AppMode.TablePairing) {
+  const [appModeState, { goToEnterPgnLink, goToTablePairingMode }] = useAppModeState();
+  const [showConfirmButton, setShowConfirm] = useState(true);
+  const [selectedPairing, setSelected] = useState<GameInfo|null>(null);
+  if (appModeState.mode !== AppMode.ArbiterSetup) {
+    return <></>;
+  }
+  if (!appModeState.pairings) {
     return <></>;
   }
+  const onSelectPairing = (pairing: GameInfo) =>{
+    if (selectedPairing === pairing){
+      setSelected(null)
+      setShowConfirm(false)
 
+    }else{
+      setSelected(pairing)
+      setShowConfirm(true)
+    }
+  }
+  const paringCardStyle = (pairing: GameInfo) => {
+    if (selectedPairing === pairing){
+      return ({
+        backgroundColor: colours.primary
+      })
+    }else{
+      return ({
+        backgroundColor: colours.secondary,
 
-  const renderPairing = ({ item }: { item: GameInfo }) => (
-    <View style={styles.pairingCard}>
-        <View style={styles.roundTextSection}>
-            <Text style={styles.roundText}>{item.round}.{item.subRound}</Text>
-        </View>
-        <View style={styles.nameTextInnerSection}>
-            <Text style={styles.nameText}>{item.players[0].firstName} {item.players[0].lastName}</Text>
-            <Text style={styles.nameText}>{item.players[1].firstName} {item.players[1].lastName}</Text>
-        </View>
-    </View>)
+      })
+    }
+  }
+  
 
+  const renderPairing = ({ item }: { item: GameInfo }) => (
+    <TouchableOpacity style={[styles.pairingCard, paringCardStyle(item)]} onPress={() => {onSelectPairing(item)}}>
+      <View style={styles.roundTextSection}>
+        <Text style={styles.roundText}>
+          {item.round}.{item.subRound}
+        </Text>
+      </View>
+      <View style={styles.nameTextInnerSection}>
+        <Text style={styles.nameText}>
+          {item.players[0].firstName} {item.players[0].lastName}
+        </Text>
+        <Text style={styles.nameText}>
+          {item.players[1].firstName} {item.players[1].lastName}
+        </Text>
+      </View>
+    </TouchableOpacity>
+  );
+  const renderConfirm = () => {
+    if(showConfirmButton && selectedPairing !== null){
+    return(<Text  onPress={() => goToTablePairingMode(selectedPairing)} style={styles.forwardBtn}>
+      Confirm {'>'}
+    </Text>)
+    }
+  }
+  
   return (
-  <View>
-      <Text onPress={returnToPgnLinkPage} style={styles.backBtn}>{'<'} Back</Text>
-      <Text style={styles.instructionTitle}>Boards</Text>
-      <Text style={styles.instructionContents}>Select the board that this iPad is assigned to. This can be changed later.</Text>
-      <FlatList scrollEnabled={true} style={styles.pairingList} data={appModeState.pairings} renderItem={renderPairing}>
+    <View>
+      <View style={styles.buttonContainer}>
+        <Text onPress={goToEnterPgnLink} style={styles.backBtn}>
+          {'<'} Back
+        </Text>
+        {renderConfirm()}
```

Comment:
Having a separate function for a specific part of a component is a bit unusual.

Usually I would write this as:
```js
...
</Text>
{showConfirmButton && selectedPairing !== null &&
  <Text  onPress={() => goToTablePairingMode(selectedPairing)} style={styles.forwardBtn}>
      Confirm {'>'}
  </Text>}
...
```

The advantage is that all of the 'presentation' logic is all at the bottom of the function in the return statement, and all of the 'business' logic is in the hooks, callbacks, etc. above. It is also more readable and concise imo.

Reply:
But also happy to hear what your thoughts are on this - this is just how *I* would do it, so not necessarily the best way...

Reply:
yep all good just couldn't figure out how to have an if section in the middle of elements